test/e2e/storage/testsuites/provisioning.go

```diff
@@ -527,6 +528,271 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
+		volumeAttachment := e2evolume.GetVolumeAttachmentName(ctx, f.ClientSet, testConfig, l.testCase.Provisioner, dataSourceRef.Name, l.sourcePVC.Namespace)
+		e2evolume.WaitForVolumeAttachmentTerminated(ctx, volumeAttachment, f.ClientSet, f.Timeouts.DataSourceProvision)
+
+		originalClaim, err := l.testCase.Client.CoreV1().PersistentVolumeClaims(l.pvc.Namespace).Get(ctx, l.sourcePVC.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Failed to get pvc: %v", err)
+
+		var originFSSize int
+		if pattern.VolMode != "Block" {
+			// Create pod to get mount path of fs
+			createdClaims := []*v1.PersistentVolumeClaim{originalClaim}
+			var podConfig *e2epod.Config = &e2epod.Config{
+				NS:            f.Namespace.Name,
+				PVCs:          createdClaims,
+				NodeSelection: testConfig.ClientNodeSelection,
+				PVCsReadOnly:  true,
+			}
+			pod, err := e2epod.CreateSecPod(ctx, l.testCase.Client, podConfig, l.testCase.Timeouts.DataSourceProvision)
+			framework.ExpectNoError(err, "Failed to create pod: %v", err)
+			mountpath := findVolumeMountPath(pod, originalClaim)
+			gomega.Expect(mountpath).ShouldNot(gomega.BeEmpty())
+
+			// Save filesystem size of the origin volume.
+			originFSSize, err = getFilesystemSizeBytes(pod, mountpath)
+			framework.ExpectNoError(err, "Failed to obtain filesystem size of a volume mount: %v", err)
+		}
+		originalSize := originalClaim.Status.Capacity.Storage()
+		biggerSize := originalSize.DeepCopy()
+		biggerSize.Add(*resource.NewQuantity(1, resource.BinarySI))
+
+		// Create claim w/ bigger size than original
+		biggerROClaim := l.testCase.Claim.DeepCopy()
+		biggerROClaim.Name = "bigger-claim"
+		biggerROClaim.Spec.Resources = v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceStorage: biggerSize,
+			},
+		}
+		biggerROClaim.Spec.AccessModes = []v1.PersistentVolumeAccessMode{
+			v1.PersistentVolumeAccessMode(v1.ReadOnlyMany),
+		}
+
+		// Create claim w/ same size as original
+		sameSizeROClaim := l.testCase.Claim.DeepCopy()
+		sameSizeROClaim.Name = "same-size-claim"
+		sameSizeROClaim.Spec.AccessModes = []v1.PersistentVolumeAccessMode{
+			v1.PersistentVolumeAccessMode(v1.ReadOnlyMany),
+		}
+
+		// For use in TestDynamicProvisioning - test expected content exists in
+		// restored volume
+		l.testCase.PvCheck = func(ctx context.Context, claim *v1.PersistentVolumeClaim) {
+			ginkgo.By("checking whether the created volume has the pre-populated data")
+			tests := []e2evolume.Test{
+				{
+					Volume:          *storageutils.CreateVolumeSource(claim.Name, false /* readOnly */),
+					Mode:            pattern.VolMode,
+					File:            "index.html",
+					ExpectedContent: expectedContent,
+				},
+			}
+			e2evolume.TestVolumeClientSlow(ctx, f, testConfig, nil, "", tests)
+		}
+
+		for _, claim := range []*v1.PersistentVolumeClaim{sameSizeROClaim, biggerROClaim} {
```

Comment:
I think this should be two different test cases (`It()`), rather than two cases under one.

Also I think from a strictly "conformant" angle, drivers that can't support expand for rox, should return an error if someone tries to create a PVC with a bigger size, so I think the bigger PVC test case should not run at all in that case.

Reply:
Are there some drivers that *can* resize while ROX? It was my understanding it was impossible since the fs is readonly. 

Re:conformance, is there a spec somewhere that dictates we should error? This is not how the PD driver [currently works](https://docs.google.com/document/d/1vYGLfKYLnfNAE1HImQ0Gp2paY2fNKexRyfXJA1jJ0tM/edit?usp=sharing) we also need to make that change and not test for size here now.

It feels like we should just remove the size comparison code here since from the comments here it now seems to be testing undefined behavior and our main motivation here was just to add *some* test for the ROX restore behavior and that's now been delayed by many months bc of confusion around sizing. I'll do that now.

Reply:
Removed file size tests

Reply:
From the CSI spec changes in https://github.com/container-storage-interface/spec/pull/452/files, it says that if the request cannot be satisfied, then the plugin should return an error.

I think it may be too late to change the PD driver to have that behavior, because it will break workflows that used to work, so what I'm proposing here is to separate it into two tests and skip the expansion case for the PD driver.

Reply:
Makes sense, but our main focus for these tests were to make sure ROX PD so I think if we want to add that it should be in a follow up/separate effort.